app/controllers/PartyTypeResolverController.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.{BaseControllerComponents, FrontendAppConfig}
+import connectors.KeystoreConnector
+import models.api.{Individual, UkCompany}
+import play.api.mvc.{Action, AnyContent}
+import services.{SessionProfile, VatRegistrationService}
+import uk.gov.hmrc.auth.core.AuthConnector
+import uk.gov.hmrc.http.InternalServerException
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class PartyTypeResolverController @Inject()(val keystoreConnector: KeystoreConnector,
```

Comment:
Looks fine, but I'd rename the class to TradingNameResolverController or similar as it's specifically for resolving which trading name page to go to. Likely my bad for referring to it so many times as a party type resolver!

Also, the method should be something like "resolve" as it doesn't show a page.